### PR TITLE
VZ-5848: Parallelize steps and stages in the upgrade paths jenkinsfile to cut down on build time

### PR DESCRIPTION
**Pull request by pshirodkar-orcl**
# Description

Parallelized few stages and functions to cut down on build time. This is first of the PRs. More PRs will follow-up later.

Fixes VZ-5848

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: ci/upgrade-paths/Jenkinsfile [hunk 1]
```diff
@@ -980,17 +980,31 @@ def upgradeVerrazzanoOnCluster(count, verrazzanoDevVersion) {
 def registerManagedClusters() {
     catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
         script {
+            // Create a dictionary of Verrazzano Register Managed Cluster steps to be executed in parallel
```
Comment (kminder) [inline ci/upgrade-paths/Jenkinsfile:983]: Is there no way to make this repeated code common?

File: ci/upgrade-paths/Jenkinsfile [hunk 2]
```diff
@@ -1090,6 +1146,46 @@ def runGinkgoRandomize(testSuitePath) {
     }
 }
 
+// Run a test suite against all clusters in parallel
+def runGinkgoRandomize(testSuitePath) {
```
Comment (kminder) [inline ci/upgrade-paths/Jenkinsfile:1150]: Should Parallel be added to the name to make it clear?

**Review by kminder — DISMISSED** [review]
Approved with comments for potential followup PR.

File: ci/upgrade-paths/Jenkinsfile [hunk 3]
```diff
@@ -454,7 +450,7 @@ pipeline {
             stages {
                 stage('verify-install') {
                     steps {
-                        runGinkgoRandomize('verify-install')
+                        runGinkgoRandomizeSerial('verify-install')
```
Comment (pshirodkar-orcl) [inline ci/upgrade-paths/Jenkinsfile:453]: `verify-install` is sort of a top-level directory containing lots of other test suites. I started seeing few failures when parallelizing. So I have kept this to be executed serially against the clusters.

File: ci/upgrade-paths/Jenkinsfile (hunk 2 above)
Comment (pshirodkar-orcl) [reply ci/upgrade-paths/Jenkinsfile]: Yes, I have added prefix Parallel to the method name now.

File: ci/upgrade-paths/Jenkinsfile (hunk 1 above)
Comment (pshirodkar-orcl) [reply ci/upgrade-paths/Jenkinsfile]: I have created a utility method for creating the map atleast. I will try to optimize further in follow-up PR.

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**